opstack/src/consensus.rs

```diff
@@ -136,6 +152,79 @@ impl Inner {
     }
 }
 
+fn verify_unsafe_signer(server_url: String, signer: Arc<Mutex<Address>>) {
+    #[cfg(not(target_arch = "wasm32"))]
+    let run = tokio::spawn;
+
+    #[cfg(target_arch = "wasm32")]
+    let run = wasm_bindgen_futures::spawn_local;
+
+    run(async move {
+        let eth_config = mainnet();
+        let mut eth_consensus = EthConsensusClient::<MainnetConsensusSpec, HttpRpc, ConfigDB>::new(
+            &eth_config
+                .consensus_rpc
+                .clone()
+                .ok_or_else(|| eyre!("missing consensus rpc"))?,
+            Arc::new(eth_config.into()),
+        )?;
+        let block = eth_consensus
+            .block_recv()
+            .unwrap()
+            .recv()
+            .await
+            .ok_or_eyre("failed to receive block")?;
+        // Query proof from op consensus server
+        let req = format!("{}signer_proof/{}", server_url, block.number.to::<u64>());
+        let proof = reqwest::get(req)
+            .await?
+            .json::<EIP1186AccountProofResponse>()
+            .await?;
+
+        // Verify unsafe signer
+        // with account proof
+        let account_path = keccak256(proof.address).to_vec();
+        let account_encoded = encode_account(&proof);
+        let is_valid = verify_proof(
+            &proof.account_proof,
+            block.state_root.as_slice(),
+            &account_path,
+            &account_encoded,
+        );
+        if !is_valid {
+            warn!(target: "helios::opstack", "account proof invalid");
+            return Err(eyre!("account proof invalid"));
+        }
+        // with storage proof
+        let storage_proof = proof.storage_proof[0].clone();
+        let key = storage_proof.key.0;
```

Comment:
We need to ensure that the key is actually the unsafe signer location. As implemented now the server can return any slot in the contract and it will be interpreted as the signer.

Reply:
Ah yes. Changing that